backend/inspirehep/orcid/domain_models.py

```diff
@@ -256,7 +252,7 @@ def _push_work_with_clashing_identifier(self):
 
             if not putcode or not recid:
                 continue
-            if recid == self.recid:
+            if int(recid) == self.recid:
```

Comment:
how do you know `self.recid` is an `int`? it's whatever type is passed to the `__init__` method, you should also make it an `int` when setting `self.recid` there.

Reply:
The point is that some tests are passing text as a wrong recid, so it cause an error.

Reply:
then you need to change those tests